overlord/configstate/config/transaction.go

```diff
@@ -343,6 +400,117 @@ func commitChange(pristine *json.RawMessage, change interface{}) *json.RawMessag
 	panic(fmt.Errorf("internal error: unexpected configuration type %T", change))
 }
 
+// partOfVirtualConfiguration() return true if the requested key
+// is part of a virtual configuration. E.g.:
+// "true" for requestedKey: "a.virtual.b" when virtualKey is "a.virtual".
+func partOfVirtualConfiguration(virtualKey, requestedKey string) (bool, error) {
+	virtualSubkeys, err := ParseKey(virtualKey)
+	if err != nil {
+		return false, err
+	}
+	requestedSubkeys, err := ParseKey(requestedKey)
+	if err != nil {
+		return false, err
+	}
+	for i := range requestedSubkeys {
+		if i >= len(virtualSubkeys) {
```

Comment:
you could move this out of the `for` loop. If you did that, you could use the same keysHaveCommonPrefix() function above, once you define it to be callable from here too.

Reply:
Would something like
```
if strings.HasPrefix(virtualKey, requestedKey+".") {
```
work here as well?